pkg/controller/v1alpha2/applicationconfiguration/component.go

```diff
@@ -36,6 +39,9 @@ func (c *ComponentHandler) Create(evt event.CreateEvent, q workqueue.RateLimitin
 		// No revision created, return
 		return
 	}
+	if err := c.cleanupControllerRevision(evt.Meta, evt.Object); err != nil {
```

Comment:
we only need to do this when we create a revision right? So why not either embed the cleanup logic there or let createControllerRevision return the revision Name to clean? We don't need to list and sort every time